spec/mixlib/install/backend/package_router_spec.rb

```diff
@@ -125,6 +125,23 @@
     end
   end
 
+  context "for a 32 bit version that is mapped to i686" do
+    let(:channel) { :stable }
+    let(:product_name) { "chef" }
+    let(:product_version) { "12.4.1" }
+    let(:platform) { "el" }
+    let(:platform_version) { "5" }
+    let(:architecture) { "i386" }
```

Comment:
If that's what you want then there is a lack of understanding what `normalize_architecture` does. It run post-query. If we request i386 (our normalized data) the omnibus.architecture property may be i686. Then we remap it back to i386.  

If we want mixlib-install to accept different architecture inputs then we have a bigger refactor.

This comes down to requirements. What are we trying to solve?